hashicorp/go-tfe
# Audit trail client

## Description

The SDK currently doesn't support the audit trails API for Terraform Cloud (I'm working on a tool which will extract these so they can be stored offline).

On the face of it I can understand why this might not be here (yet):
- The audit trails API doesn't adhere to JSONAPI as the rest of the calls seem to. This means I can't rely on the default response marshaling, but fortunately someone was kind enough to allow the use of an io.Writer so I can bypass this and do it manually.
- The audit trails API requires an org token instead of a regular API token

## Testing plan

- Run a simple test with an org token to see if we can access audit trails
- Make sure that the list is non-zero in length (because even with an api token you _do_ get a response - just no results)

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [API documentation](https://www.terraform.io/docs/cloud/api/audit-trails.html)

## Output from tests

I'm about to add the tests, just wanted to see if this feature is something that you would merge, or would rather not have in this SDK due to the reasons I've highlighted.
